Src/Newtonsoft.Json-for-Unity.Tests/Assets/SystemWebPolyfill/Util/HttpEncoder.cs

```diff
@@ -0,0 +1,796 @@
+                return null;
+
+            if (s.Length == 0)
+                return String.Empty;
+            
+            if (s.IndexOf ('&') == -1)
+                return s;
+            StringBuilder rawEntity = new StringBuilder ();
+            StringBuilder entity = new StringBuilder ();
+            StringBuilder output = new StringBuilder ();
+            int len = s.Length;
+            // 0 -> nothing,
+            // 1 -> right after '&'
+            // 2 -> between '&' and ';' but no '#'
+            // 3 -> '#' found after '&' and getting numbers
+            int state = 0;
+            int number = 0;
+            bool is_hex_value = false;
+            bool have_trailing_digits = false;
+    
+            for (int i = 0; i < len; i++) {
+                char c = s [i];
+                if (state == 0) {
+                    if (c == '&') {
+                        entity.Append (c);
+                        rawEntity.Append (c);
+                        state = 1;
+                    } else {
+                        output.Append (c);
+                    }
+                    continue;
+                }
+
+                if (c == '&') {
+                    state = 1;
+                    if (have_trailing_digits) {
+                        entity.Append (number.ToString (Helpers.InvariantCulture));
+                        have_trailing_digits = false;
+                    }
+
+                    output.Append (entity.ToString ());
+                    entity.Length = 0;
+                    entity.Append ('&');
+                    continue;
+                }
+
+                if (state == 1) {
+                    if (c == ';') {
+                        state = 0;
+                        output.Append (entity.ToString ());
+                        output.Append (c);
+                        entity.Length = 0;
+                    } else {
+                        number = 0;
+                        is_hex_value = false;
+                        if (c != '#') {
+                            state = 2;
+                        } else {
+                            state = 3;
+                        }
+                        entity.Append (c);
+                        rawEntity.Append (c);
+                    }
+                } else if (state == 2) {
+                    entity.Append (c);
+                    if (c == ';') {
+                        string key = entity.ToString ();
+                        if (key.Length > 1 && Entities.ContainsKey (key.Substring (1, key.Length - 2)))
+                            key = Entities [key.Substring (1, key.Length - 2)].ToString ();
+
+                        output.Append (key);
+                        state = 0;
+                        entity.Length = 0;
+                        rawEntity.Length = 0;
+                    }
+                } else if (state == 3) {
+                    if (c == ';') {
+                        if (number == 0)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/jilleJr/Newtonsoft.Json-for-Unity/pullRequest?prid=4755183)